tests/unit_tests/test_analytics/test_sentry.py

```diff
@@ -0,0 +1,602 @@
+import os
+from unittest import mock
+
+import pytest
+import wandb
+import wandb.analytics
+import wandb.env
+from sentry_relay import MetricRelayServer
+from sentry_sdk.utils import sentry_sdk
+from tests.unit_tests.test_analytics.sentry_relay import SentryResponse
+
+SENTRY_DSN_FORMAT = "http://{key}@127.0.0.1:{port}/{project}"
+
+
+@pytest.fixture(scope="module")
+def relay():
+    relay_server = MetricRelayServer()
+    relay_server.start()
+
+    yield relay_server
+
+    relay_server.stop()
+
+
+def test_wandb_sentry_does_not_interfer_with_global_sentry_sdk(
+    relay: MetricRelayServer,
+):
+    """
+    Test that wandb sentry initialization does not interfere with global sentry_sdk.
+    """
+    with mock.patch.dict(
+        os.environ,
+        {
+            wandb.env.ERROR_REPORTING: "true",
+            wandb.env.SENTRY_DSN: SENTRY_DSN_FORMAT.format(
+                key="WANDB_SENTRY_PUBLIC_KEY",
+                port=relay.port,
+                project="123456",
+            ),
+        },
+    ):
+        sentry_sdk.init(
+            dsn=SENTRY_DSN_FORMAT.format(
+                key="OTHER_SENTRY_PUBLIC_KEY",
+                port=relay.port,
+                project="654321",
+            ),
+            default_integrations=False,
+        )
+        wandb_sentry = wandb.analytics.Sentry()
+        wandb_sentry.setup()
+
+        # Assert wandb Sentry scope and dsn are different from the other Sentry client
+        assert sentry_sdk.get_current_scope() != wandb_sentry.scope
+        assert (
+            sentry_sdk.get_current_scope().client.dsn != wandb_sentry.scope.client.dsn  # type: ignore
+        )
+
+
+def test_wandb_error_reporting_disabled(relay: MetricRelayServer):
+    """
+    This test ensures no events are sent to sentry when the wandb sentry client is disabled.
+
+    The test sets the `ERROR_REPORTING` environment variable to `false` and initializes the wandb sentry client.
+
+    This test replaces the landfill test verifying disabling of wandb Sentry client via environment variable.
+        - t6_sentry_third_party_no_wandb_error_reporting.py
```

Comment:
sorry i meant just leave a comment on the sentry file in github
(see the example i left above)